velox/type/Conversions.h

```diff
@@ -383,6 +383,38 @@ struct Converter<TypeKind::TIMESTAMP> {
   static T cast(const std::string& v, bool& nullOutput) {
     return fromTimestampString(v);
   }
+
+  static T cast(Date& d, bool& nullOutput) {
+    static const int32_t kMillisPerDay{86400000};
+    return Timestamp::fromMillis(d.days() * kMillisPerDay);
+  }
+};
+
+// Allow conversions from string to DATE type.
+template <>
+struct Converter<TypeKind::DATE> {
+  using T = typename TypeTraits<TypeKind::DATE>::NativeType;
+  template <typename From>
+  static T cast(const From& /* v */, bool& nullOutput) {
+    VELOX_NYI();
+  }
+
+  static T cast(folly::StringPiece v, bool& nullOutput) {
+    return fromDateString(v.data(), v.size());
+  }
+
+  static T cast(const StringView& v, bool& nullOutput) {
+    return fromDateString(v);
+  }
+
+  static T cast(const std::string& v, bool& nullOutput) {
+    return fromDateString(v);
+  }
+
+  static T cast(Timestamp& t, bool& nullOutput) {
+    static const int32_t kSecsPerDay{86400};
```

Comment:
nit: kSecsPerDay -> kSecondsPerDay{86'400} for readability and to follow "don't abbreviate" guideline.